api/provider/aws/service_list.go

```diff
@@ -12,102 +12,78 @@ import (
+	fn := func(output *ecs.ListServicesOutput, lastPage bool) bool {
+		for _, serviceARN := range output.ServiceArns {
+			// sample service ARN:
+			// arn:aws:ecs:us-west-2:856306994068:service/l0-tlakedev-guestbo80d9d
+			serviceName := strings.Split(aws.StringValue(serviceARN), "/")[1]
+			serviceNames = append(serviceNames, serviceName)
 		}
 
-		clusterServices[clusterArn] = serviceArns
+		return !lastPage
 	}
 
-	serviceSummaries := []models.ServiceSummary{}
-	for _, serviceArns := range clusterServices {
-		for _, serviceArn := range serviceArns {
-			ecsServiceID := s.serviceARNToECSServiceID(aws.StringValue(serviceArn))
-			service := &models.Service{
-				ServiceID: ecsServiceID,
-			}
-			s.populateSummariesTags(service, ecsServiceID)
-
-			summary := models.ServiceSummary{
-				ServiceID:       service.ServiceID,
-				ServiceName:     service.ServiceName,
-				EnvironmentID:   service.EnvironmentID,
-				EnvironmentName: service.EnvironmentName,
-			}
-
-			serviceSummaries = append(serviceSummaries, summary)
+	for _, clusterName := range clusterNames {
+		input := &ecs.ListServicesInput{}
+		input.SetCluster(clusterName)
+		if err := s.AWS.ECS.ListServicesPages(input, fn); err != nil {
+			return nil, err
 		}
 	}
 
-	return serviceSummaries, nil
+	return serviceNames, nil
 }
 
-func (s *ServiceProvider) listServices(clusterArn *string) ([]*string, error) {
-	serviceArns := []*string{}
-	input := &ecs.ListServicesInput{
-		Cluster: clusterArn,
+func (s *ServiceProvider) makeServiceSummaryModels(serviceIDs []string) ([]models.ServiceSummary, error) {
+	serviceTags, err := s.TagStore.SelectByType("service")
+	if err != nil {
+		return nil, err
 	}
 
-	for {
-		output, err := s.AWS.ECS.ListServices(input)
-		if err != nil {
-			return nil, err
-		}
-
-		serviceArns = append(serviceArns, output.ServiceArns...)
-		input.NextToken = output.NextToken
+	models := make([]models.ServiceSummary, len(serviceIDs))
+	for i, serviceID := range serviceIDs {
+		models[i].ServiceID = serviceID
 
-		if output.NextToken == nil {
-			break
+		if tag, ok := serviceTags.WithID(serviceID).WithKey("name").First(); ok {
+			models[i].ServiceName = tag.Value
 		}
-	}
-
-	return serviceArns, nil
-}
 
-func (s *ServiceProvider) listClusters() ([]*string, error) {
-	input := &ecs.ListClustersInput{}
-	clusterArns := []*string{}
+		var environmentID string
+		if tag, ok := serviceTags.WithID(serviceID).WithKey("environment_id").First(); ok {
+			environmentID = tag.Value
+			models[i].EnvironmentID = environmentID
+		}
 
-	for {
-		output, err := s.AWS.ECS.ListClusters(input)
+		environmentTags, err := s.TagStore.SelectByTypeAndID("environment", environmentID)
 		if err != nil {
 			return nil, err
 		}
 
-		clusterArns = append(clusterArns, output.ClusterArns...)
-		input.NextToken = output.NextToken
-
-		if output.NextToken == nil {
-			break
+		if tag, ok := environmentTags.WithKey("name").First(); ok {
+			models[i].EnvironmentName = tag.Value
 		}
 	}
 
```

Comment:
you need to populate the load balancer id and name tag

Reply:
ignore me

Reply:
xD ✅